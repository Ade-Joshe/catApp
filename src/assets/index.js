export * from "./allCatIcon";
export * from "./catsILikeIcon";
export * from "./emptyHeart";
export * from "./filledHeart";