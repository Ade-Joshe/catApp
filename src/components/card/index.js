export * from "./miniCard";
export * from "./emptyCard";
export * from "./largeCard";