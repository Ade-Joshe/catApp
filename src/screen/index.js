export * from "./favouriteScreen";
export * from "./homeScreen";