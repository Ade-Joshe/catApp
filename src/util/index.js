export * from "./pixelConverter";
export * from "./theme";
export * from "./store";