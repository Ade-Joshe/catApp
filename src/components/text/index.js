import React, { useContext } from 'react'
import { StyleSheet, Text } from 'react-native'
import { fontPixel, heightPixel, ThemeContext } from '../../util'

export const PageTitle = ({ children }) => {
    const { colors } = useContext(ThemeContext);
    return <Text style={[styles.title, { color: colors.black, }]}>{children}</Text>
};

const styles = StyleSheet.create({
    title: { fontSize: fontPixel(16), marginVertical: heightPixel(25), fontWeight: "600" }
})
