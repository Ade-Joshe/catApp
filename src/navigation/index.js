import React, { useContext } from "react"
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from "@react-navigation/native";
import { HomeScreen, FavouriteScreen } from '../screen';
import { AllCatIcon, CatsILikeIcon } from "../assets";
import { fontPixel, ThemeContext } from "../util";

const Tab = createBottomTabNavigator();

export function RootNavigation() {

    const { colors } = useContext(ThemeContext);

    return (
        <NavigationContainer>
            <Tab.Navigator
                screenOptions={({ route }) => ({
                    unmountOnBlur: true,
                    headerShown: false,
                    lazy: true,
                    tabBarActiveTintColor: colors.black,
                    tabBarInactiveTintColor: colors.grey,
                    tabBarStyle: { height: 67 },
                    tabBarLabelStyle: { fontSize: fontPixel(15) },
                    tabBarLabelPosition: 'below-icon',
                    tabBarIcon: ({ focused }) => {
                        if (route.name === 'All Cats') return <AllCatIcon color={focused ? colors.black : colors.grey} />;
                        else if (route.name === 'Cats I Like') return <CatsILikeIcon color={focused ? colors.black : colors.grey} />;
                        else return;
                    }
                })}
            >
                <Tab.Screen name="All Cats" component={HomeScreen} />
                <Tab.Screen name="Cats I Like" component={FavouriteScreen} />
            </Tab.Navigator>
        </NavigationContainer >
    );
}