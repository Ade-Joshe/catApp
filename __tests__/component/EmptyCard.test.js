import React from "react"
import { render } from '@testing-library/react-native';
import { EmptyCard } from "../../src/components";
import { create } from "react-test-renderer";

// // renders loading
it("renders correctly when loading is passed", () => {
    const { getAllByText } = render(<EmptyCard status={"loading"} />);

    expect(getAllByText("Loading...")).toBeTruthy();
});

// // renders error
it("renders correctly when error is passed", () => {
    const { getAllByText } = render(<EmptyCard status={"error"} />);

    expect(getAllByText("No data found!")).toBeTruthy();
});

// // renders  v
it("renders correctly when error is passed", () => {
    const { getAllByText } = render(<EmptyCard status={"emptyFav"} />);

    expect(getAllByText("We keep your favouites here")).toBeTruthy();
});

it("snapshot testing for matching loading", () => {
    let component = create(<EmptyCard status={"loading"} />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

it("snapshot testing for matching error", () => {
    let component = create(<EmptyCard status={"error"} />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

it("snapshot testing for matching emptyFav", () => {
    let component = create(<EmptyCard status={"emptyFav"} />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});