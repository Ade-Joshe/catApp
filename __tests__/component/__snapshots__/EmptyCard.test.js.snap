// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for matching emptyFav 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#21222790",
        "fontSize": 25,
        "marginTop": "40%",
        "marginVertical": 17,
      }
    }
  >
    We keep your favouites here
  </Text>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="18"
    bbWidth="18"
    fill="none"
    focusable={false}
    height="18"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 18,
          "width": 18,
        },
      ]
    }
    vbHeight={18}
    vbWidth={18}
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M15.63 3.458a4.124 4.124 0 00-5.835 0L9 4.253l-.795-.795A4.126 4.126 0 002.37 9.293l.795.795L9 15.922l5.835-5.835.795-.795a4.124 4.124 0 000-5.835z"
        fill={4292739586}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={4292739586}
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="1.5"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`snapshot testing for matching error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#212227",
        "fontSize": 25,
        "marginTop": "40%",
        "marginVertical": 17,
      }
    }
  >
    No data found!
  </Text>
</View>
`;

exports[`snapshot testing for matching loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size={60}
    style={
      Object {
        "marginTop": "40%",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#212227",
        "fontSize": 25,
        "marginVertical": 17,
      }
    }
  >
    Loading...
  </Text>
</View>
`;
